src/golang/cmd/server/request/parser/dag_id.go

```diff
@@ -0,0 +1,26 @@
+package parser
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/aqueducthq/aqueduct/cmd/server/routes"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/google/uuid"
+)
+
+type DagIDParser struct{}
+
+func (DagIDParser) Parse(r *http.Request) (uuid.UUID, error) {
```

Comment:
Can we just have a generic ID parser instead of defining one for each ID type

Reply:
This is actually something inspired by OpenAPI that each parameter has its own definition, so that I feel it makes sense to create one specific parser in application logic. In this way the caller only need to worry about 'parsing a parameter'. I think 'type parsing' is a lower level abstraction which we can certainly iterate further, but I don't see a strong motivation getting that abstraction refined now

Reply:
Reference: https://github.com/aqueducthq/aqueduct/pull/1186/files look for `parameters` in `aqueduct.json`